proto/zitadel/user/v3alpha/user_service.proto

```diff
@@ -0,0 +1,2049 @@
+  // because of an event that happened (wrong password, etc.)
+  //
+  // The endpoint returns an error if the user is already in the state 'locked'.
+  rpc LockUser (LockUserRequest) returns (LockUserResponse) {
+    option (google.api.http) = {
+      post: "/v3alpha/users/{user_id}/lock"
+    };
+
+    option (zitadel.protoc_gen_zitadel.v2.options) = {
+      auth_option: {
+        permission: "authenticated"
+      }
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      responses: {
+        key: "200";
+        value: {
+          description: "User successfully locked";
+        };
+      };
+    };
+  }
+
+  // Unlock a user
+  //
+  // Unlock a previously locked user and change the state to 'active'.
+  // The user will be able to log in again.
+  //
+  // The endpoint returns an error if the user is not in the state 'locked'.
+  rpc UnlockUser (UnlockUserRequest) returns (UnlockUserResponse) {
+    option (google.api.http) = {
+      post: "/v3alpha/users/{user_id}/unlock"
+    };
+
+    option (zitadel.protoc_gen_zitadel.v2.options) = {
+      auth_option: {
+        permission: "authenticated"
+      }
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      responses: {
+        key: "200";
+        value: {
+          description: "User successfully unlocked";
+        };
+      };
+    };
+  }
+
+  // Delete a user
+  //
+  // Delete an existing user and change the state to 'deleted'.
+  // The user will be able to log in anymore.
+  rpc DeleteUser (DeleteUserRequest) returns (DeleteUserResponse) {
+    option (google.api.http) = {
+      delete: "/v3alpha/users/{user_id}"
+    };
+
+    option (zitadel.protoc_gen_zitadel.v2.options) = {
+      auth_option: {
+        permission: "authenticated"
+      }
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      responses: {
+        key: "200";
+        value: {
+          description: "User successfully deleted";
+        };
+      };
+    };
+  }
+
+  // Validate a user
+  //
+  // Verify if an existing user satisfies a schema.
+  // This can be used if a user is based on an older revision of the schema
+  // or if a user needs to be migrated to a new schema.
+  rpc ValidateUser (ValidateUserRequest) returns (ValidateUserResponse) {
+    option (google.api.http) = {
+      post: "/v3alpha/users/{user_id}/validate"
+      body: "*"
+    };
+
+    option (zitadel.protoc_gen_zitadel.v2.options) = {
+      auth_option: {
+        permission: "authenticated"
+      }
+    };
+
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      responses: {
+        key: "200"
+        value: {
+          description: "User validation completed"; //(--TODO: do we always return 200/no error if the process was ok with potential validation error list or a 400?--)
+        }
+      };
+    };
+  }
```

Comment:
do we always return 200/no error if the process was ok with potential validation error list or a 400?

Reply:
I would like an error if one of the validations failed, but at the moment as you could have multiple errors a 200 and a list of errors would work better, as the validation was OK but the contents could not be handled.

Reply:
exactly. what i'm trying to figure out is, how to tell the caller the difference between a wrong argument / precondition on his side or for the validation and possible validation errors.
A 200 with a list would make that distinction easy, but I'm generally no fan of 200 with errors

Reply:
I agree, thanks to SAML 👎 which handles it like that, we could also split it into 2 endpoints "Check" and "Validate", where "Check" gives the first error and "Validate" handles it with the 200 with errors?

Reply:
as just discussed, since the client anyway needs to handle this / can validate it on his side, we omit the call as of now